### PR TITLE
Initialise submodules on demand

### PR DESCRIPTION
**Pull request by wom-bat**
Add dependencies so that Make knows that submodules need to be downloaded, and add rules to download them.

### CODE REVIEW

File: examples/vmm/vmm-dev.mk [hunk 1]
```diff
@@ -124,3 +128,11 @@ clobber:: clean
 
 # How to build libvmm.a
 include ${LIBVMM}/vmm.mk
+
+# Add an extra dependency for libvmm.a
+# Note: this should be in ${LIBVMM}/vmm.mk
+libvmm/virtio/block.o: ${SDDF}/include
```
Comment (Ivan-Velickovic) [inline examples/vmm/vmm-dev.mk:134]: I noticed when working on libvmm that updating the headers did not cause objects to be rebuilt, I think we should add this as well as `libvmm/include` to `libvmm/vmm.mk` in another PR.

Comment (wom-bat) [reply examples/vmm/vmm-dev.mk:134]: libvmm/include should not be necessary providing -MD is in CFLAGS.

Comment (wom-bat) [reply examples/vmm/vmm-dev.mk:134]: I've just looked at vmm.mk --- it has a different bug, stemming from when object files were put in a libvmm directory, not a src directory -- the result is the dependency files that track headers were never included.
PR for libvmm coming soon.

Comment (Ivan-Velickovic) [reply examples/vmm/vmm-dev.mk:134]: Still seems weird that we have this line. Do we need it anymore after the fix to libvmm?

**Comment by Ivan-Velickovic** [comment]
@JE-Archer ran into an issue where he manually updated a submodule dependency (sDDF) and running `make` kept resetting it to the old commit.

Another problem I've just realised is the scenario where someone's got an old checkout of LionsOS, goes to an example and runs `git pull` and then `make`. None of the submodules will be updated since `git pull` doesn't do that but there is also no `make submodules` command they can run to update the submodules.

It might be worth going back to having an explicit `make submodules` command.

**Pull request merged**